trinity/plugins/builtin/json_rpc/plugin.py

```diff
@@ -48,15 +46,20 @@ def do_start(self) -> None:
         db_manager = create_db_manager(self.context.trinity_config.database_ipc_path)
         db_manager.connect()
 
-        chain_class = self.context.trinity_config.node_class.chain_class
+        trinity_config = self.context.trinity_config
+        chain_config = trinity_config.get_chain_config()
+
+        chain: BaseAsyncChain
 
-        if self.context.trinity_config.sync_mode == SYNC_LIGHT:
+        if self.context.trinity_config.is_light_mode:
             header_db = db_manager.get_headerdb()  # type: ignore
             event_bus_light_peer_chain = EventBusLightPeerChain(self.context.event_bus)
-            chain = chain_class(header_db, peer_chain=event_bus_light_peer_chain)
-        else:
+            chain = chain_config.light_chain_class(header_db, peer_chain=event_bus_light_peer_chain)
+        elif trinity_config.is_full_mode:
```

Comment:
>If we're looking for more type-safety, sync_mode could be an `Enum` value instead of a `str`.

With just two options, I don't feel strongly here but in general I agree that an `Enum` is nicer. Will change that.

Reply:
@carver I found myself needing to import the enum/constant anywhere that I wanted to check sync mode, thus arose the creation of the `is_x_mode`.  I don't feel strongly, but it was done to clean up the call sites.